### ISSUE TITLE
VSCode's ESLint extension server crash - `Cannot redefine plugin "test"`

### ISSUE THREAD

**Issue by NamesMT** [antfu/vitesse-nuxt#111]
### Describe the bug

VSCode's ESLint extension server crashed after the first on save fix.

The issue seems to be/happen when `@nuxt/eslint` is combined with `@antfu/eslint-config`.
Idk if this is the right place to create the issue, but this repo seems like a good choice because the current repo is using the mentioned setup, with the issue at present.

The issue happens to this template repo as-is,
The issue happens to any Nuxt instance: setting up a fresh Nuxt instance with `nuxi init`, installing `@nuxt/eslint` + `@antfu`'s config following the tutorial at [eslint.nuxt.com](https://eslint.nuxt.com/packages/module#custom-config-presets), the same issue can be observed.

Reproduce:
Simply clone this repo, install deps, add some blank linebreaks to a file then save, after the first auto-correction of the blank linebreaks being removed, ESLint extension's server will crash and no longer auto-corrects on save.

Image:
![image](https://github.com/antfu/vitesse-nuxt3/assets/23612546/10b9179c-cc28-47c9-8ef4-b862946abeda)

Some additional contexts:
Running ESLint from CLI seems to work normally (using both `@nuxt/eslint` and `@antfu`).
ESLint works in other repos where only `@antfu`'s config is used.
ESLint works for Nuxt when using only `@nuxt/eslint`.
Because ESLint from CLI works normally, I tried replacing VSCode's bundled NodeJS (v18), which is used by VSCode to run the ESLint server to latest LTS (v20) which I'm using for CLI, the same issue is observed, so not related to NodeJS version differences.

### Reproduction

https://stackblitz.com/github/antfu/vitesse-nuxt3

### System Info

```Shell
System:
    OS: Linux 5.15 Alpine Linux
    CPU: (16) x64 AMD Ryzen 7 5700U with Radeon Graphics
    Memory: 3.33 GB / 6.70 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.12.0 - ~/.nvm/versions/node/v20.12.0/bin/node
    Yarn: 1.22.22 - ~/.local/share/pnpm/yarn
    npm: 10.5.0 - ~/.nvm/versions/node/v20.12.0/bin/npm
    pnpm: 8.15.5 - ~/.local/share/pnpm/pnpm
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by NamesMT** [comment]
FYI: using `@antfu` as the export base and `@nuxt` as a userConfig works fine and doesn't crash the ESLint extension server.
(Stylistic silencing disabled).
![Before autofix](https://github.com/antfu/vitesse-nuxt3/assets/23612546/ad450da7-b8d0-4936-bfd5-e1b682d0f34e)
![After autofix](https://github.com/antfu/vitesse-nuxt3/assets/23612546/6be6377b-1a8d-410e-b3c6-be85c84a8ba1)

I don't know how much behavior differences there are though. (as now `@nuxt/eslint` is overriding `@antfu` instead of vice-versa)
Update:
Good news for any experiencing the same problem as me, using `@eslint/config-inspector` (also made by Anthony), it seems like there is currently no behavior difference (in the current state), so we could use antfu as the export base for the current workaround.
The only duplicated rule in the two configs are `vue/multi-word-component-names`, but is both turned off.

**Comment by mirari** [comment]
Update:  `"@nuxt/eslint": "^0.3.0-beta.7",` fixes the following mentioned problem, but hasn't been updated to `vitesse-nuxt3` yet

---

This new config will cause eslint to fail under windows os:
```
Oops! Something went wrong! :(

ESLint: 8.57.0

Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'd:'
    at throwIfUnsupportedURLScheme (node:internal/modules/esm/load:239:11)
    at defaultLoad (node:internal/modules/esm/load:130:3)
    at ModuleLoader.load (node:internal/modules/esm/loader:403:13)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:285:56)
    at new ModuleJob (node:internal/modules/esm/module_job:67:26)
    at #createModuleJob (node:internal/modules/esm/loader:297:17)
    at ModuleLoader.getJobFromResolveResult (node:internal/modules/esm/loader:254:34)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:235:17)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:21)

```
You can reproduce this by pulling the code and executing the lint command in windows.

It seems that there is a compatibility issue with `eslint.config.mjs` in the `.nuxt` directory:
```
import { pipe } from 'D:/dir/work/project/node_modules/.pnpm/eslint-flat-config-utils@0.1.2/node_modules/eslint-flat-config-utils/dist/index.mjs';
import typegen from 'D:/dir/work/project/node_modules/.pnpm/eslint-typegen@0.1.6/node_modules/eslint-typegen/dist/index.mjs';
import { createConfigForNuxt, defineFlatConfigs } from 'D:/dir/work/project/node_modules/.pnpm/@nuxt+eslint-config@0.3.0-beta.6_eslint@8.57.0_typescript@5.4.3/node_modules/@nuxt/eslint-config/dist/flat.mjs';
```
Until the March 27th update, this file will not contain these references.

**Comment by antfu** [comment]
Is this still happening?

**Comment by NamesMT** [comment]
Hi @antfu, with all deps updated to latest version in [my repo](https://github.com/NamesMT/starter-nuxt/tree/a/v-n3%23111), the issue still exists:
```
[Error - 3:29:26 PM] An unexpected error occurred:
[Error - 3:29:26 PM] ConfigError: Config "antfu/test/setup": Key "plugins": Cannot redefine plugin "test".
    at rethrowConfigError (/home/mine/starter-nuxt/node_modules/.pnpm/@humanwhocodes+config-array@0.12.3/node_modules/@humanwhocodes/config-array/api.js:225:8)
    at /home/mine/starter-nuxt/node_modules/.pnpm/@humanwhocodes+config-array@0.12.3/node_modules/@humanwhocodes/config-array/api.js:1018:5
    at Array.reduce (<anonymous>)
    at FlatConfigArray.getConfig (/home/mine/starter-nuxt/node_modules/.pnpm/@humanwhocodes+config-array@0.12.3/node_modules/@humanwhocodes/config-array/api.js:1014:39)
    at ESLint.calculateConfigForFile (/home/mine/starter-nuxt/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1175:24)
    at async ESLint.isPathIgnored (/home/mine/starter-nuxt/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1197:24)
    at async ESLint.lintText (/home/mine/starter-nuxt/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1034:33)
    at async /root/.vscode-server/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:24954
    at async E (/root/.vscode-server/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:19114)
    at async /root/.vscode-server/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:220290
```

The issue also exists with this repo (`vitesse-nuxt3`) right now:
```
[Error - 3:37:16 PM] An unexpected error occurred:
[Error - 3:37:16 PM] ConfigError: Config "antfu/test/setup": Key "plugins": Cannot redefine plugin "test".
    at rethrowConfigError (/home/mine/test-vn3/node_modules/.pnpm/@humanwhocodes+config-array@0.12.3/node_modules/@humanwhocodes/config-array/api.js:225:8)
    at /home/mine/test-vn3/node_modules/.pnpm/@humanwhocodes+config-array@0.12.3/node_modules/@humanwhocodes/config-array/api.js:1018:5
    at Array.reduce (<anonymous>)
    at FlatConfigArray.getConfig (/home/mine/test-vn3/node_modules/.pnpm/@humanwhocodes+config-array@0.12.3/node_modules/@humanwhocodes/config-array/api.js:1014:39)
    at ESLint.calculateConfigForFile (/home/mine/test-vn3/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1175:24)
    at async ESLint.isPathIgnored (/home/mine/test-vn3/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1197:24)
    at async ESLint.lintText (/home/mine/test-vn3/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1034:33)
    at async /root/.vscode-server/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:24954
    at async E (/root/.vscode-server/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:19114)
    at async /root/.vscode-server/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:220290
```

For more context, I'm using Windows 11 + WSL2, distros: Alpine 3.18 + Alpine 3.19.

**Comment by NamesMT** [comment]
Closed by antfu/eslint-config@2bc6f13
